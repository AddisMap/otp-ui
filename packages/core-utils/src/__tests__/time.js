import {
  restoreDateNowBehavior,
  setDefaultTestTime
} from "../../../../test-utils/time";

import {
  formatDuration,
  formatDurationWithSeconds,
  getCurrentDate,
  getCurrentTime
} from "../time";

describe("time", () => {
  afterEach(restoreDateNowBehavior);

  describe("getCurrentDate", () => {
    it("should return current date at specified timezone", () => {
      setDefaultTestTime();
      expect(getCurrentDate("America/New_York")).toMatchSnapshot();
    });
  });

  describe("getCurrentTime", () => {
    it("should return time at specified timezone", () => {
      setDefaultTestTime();
      expect(getCurrentTime("America/New_York")).toMatchSnapshot();
    });
  });

  describe("getFormattedTime", () => {
    const durationInSeconds = 9401;
    it(`should correctly format ${durationInSeconds} seconds as a duration with seconds`, () => {
      expect(formatDurationWithSeconds(durationInSeconds)).toMatchSnapshot();
    });
    it(`should correctly format ${durationInSeconds} seconds as a duration without seconds`, () => {
      expect(formatDuration(durationInSeconds)).toMatchSnapshot();
    });
  });
});
