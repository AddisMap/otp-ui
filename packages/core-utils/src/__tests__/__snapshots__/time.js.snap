// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time getCurrentDate should return current date at specified timezone 1`] = `"2019-08-04"`;

exports[`time getCurrentTime should return time at specified timezone 1`] = `"22:34"`;

exports[`time getFormattedTime should correctly format 9401 seconds as a duration with seconds 1`] = `"2 hr 36 min 41 sec"`;

exports[`time getFormattedTime should correctly format 9401 seconds as a duration without seconds 1`] = `"2 hr 36 min"`;
